qcodes/dataset/data_set.py

```diff
@@ -944,10 +944,20 @@ def get_parameter_data(
         The values are returned as a dictionary with names of the requested
         parameters as keys and values consisting of dictionaries with the
         names of the parameters and its dependencies as keys and numpy arrays
-        of the data as values. If some of the parameters are stored as arrays
+        of the data as values. If the dataset has a shape recorded
+        in its metadata and the number of datapoints recorded matches the
+        expected number of points the data will be returned as numpy arrays
+        in this shape. If there are less datapoints recorded than expected
+        from the metadata the dataset will be returned as is. This could happen
+        if you call `get_parameter_data` on an incomplete dataset. See
+        :py:class:`.DataSetCache.data` for an implementation that returns the
```

Comment:
would `DataSet.cache.data` work as a better reference? i'm just a bit worried that users would get confused in a "oh, i need some other class, oh dear" way. perhaps i'm too careful)